src/atomate2/abinit/sets/base.py

```diff
@@ -0,0 +1,560 @@
+from abipy.flowtk.psrepos import get_repo_from_name
+from abipy.flowtk.utils import Directory, irdvars_for_ext
+from monty.json import MSONable
+from pymatgen.core.structure import Structure
+from pymatgen.io.abinit.pseudos import PseudoTable
+from pymatgen.io.core import InputGenerator, InputSet
+
+from atomate2.abinit.files import (
+    fname2ext,
+    load_abinit_input,
+    load_generator,
+    out_to_in,
+)
+from atomate2.abinit.utils.common import (
+    INDATA_PREFIX,
+    INDATAFILE_PREFIX,
+    INDIR_NAME,
+    INPUT_FILE_NAME,
+    OUTDATA_PREFIX,
+    OUTDATAFILE_PREFIX,
+    OUTDIR_NAME,
+    TMPDATA_PREFIX,
+    TMPDIR_NAME,
+    InitializationError,
+)
+
+__all__ = ["AbinitInputSet", "AbinitInputSetGenerator"]
+
+logger = logging.getLogger(__name__)
+
+
+class AbinitInputSet(InputSet):
+    """
+    A class to represent a set of Abinit inputs.
+
+    Parameters
+    ----------
+    abinit_input
+        An AbinitInput object.
+    """
+
+    def __init__(
+        self,
+        abinit_input: AbinitInput,
+        input_files: Optional[Iterable[Union[str, Path, dict]]] = None,
+        link_files: bool = True,
+        validation: bool = True,
+        generator: Optional[MSONable] = None,
+    ):
+        self.input_files = input_files
+        self.link_files = link_files
+        self.validation = validation
+        self.generator = generator
+        super().__init__(inputs={INPUT_FILE_NAME: abinit_input})
+
+    def write_input(
+        self,
+        directory: Union[str, Path],
+        make_dir: bool = True,
+        overwrite: bool = True,
+        zip_inputs: bool = False,
+    ):
+        """Write Abinit input files to a directory."""
+        # TODO: do we allow zipping ? not sure if it really makes sense for abinit as
+        #  the abinit input set also sets up links to previous files, sets up the
+        #  indir, outdir and tmpdir, ...
+        self.inputs["abinit_input.json"] = json.dumps(self.abinit_input.as_dict())
+        if self.generator:
+            gendict = self.generator.as_dict()
+            self.inputs["abinit_input_set_generator.json"] = json.dumps(gendict)
+        super().write_input(
+            directory=directory,
+            make_dir=make_dir,
+            overwrite=overwrite,
+            zip_inputs=zip_inputs,
+        )
+        del self.inputs["abinit_input.json"]
+        if self.generator:
+            del self.inputs["abinit_input_set_generator.json"]
+        self.indir, self.outdir, self.tmpdir = self.set_workdir(workdir=directory)
+
+        if self.input_files:
+            out_to_in(
+                out_files=self.input_files,
+                indir=self.indir.path,
+                link_files=self.link_files,
+            )
+
+        if self.validation:
+            self.validate()
+
+    def validate(self):
+        # Check that all files in the input directory have their corresponding
+        # ird variables.
+        for filename in os.listdir(self.indir.path):
```

Comment:
Actually thinking again about this, I would still keep a validation (that can be done before anyway). In Abinit, if you don't specify that you want to read a file, it is just ignored. If we pass some input files to the input set but do not set the corresponding variables to specify that you want to read these input files, it will just continue (maybe crash but not necessarily, depending on the calculation).

Reply:
Done.